Guide/scripts.markdown

```diff
@@ -64,6 +64,29 @@ This is made possible because of the [she-bang line](https://en.wikipedia.org/wi
 
 In case you get a permission error, try to add the executable flag via `chmod +x Application/Script/HelloWorldToAllUsers.hs`.
 
+## Running a script from ghci
+
+You can also open a ghci prompt to test features in scripts interactively.
+
+```bash
+make ghci
+```
+
+Then you can load your script into the interpreter:
+```
+:l Application/Scripts/MyScript
+```
+
+and run the script from the IHP ghci command line:
+
+```haskell
+IHP> runScript ihpDefaultConfig runTestName
+```
+
+The `ihpDefaultConfig` is made available from the `Application.Script.Prelude` import but can be substituted
+with your own `Config` data structure. This is particularly useful for adjusting logging levels or
```

Comment:
Maybe we can have an example call to `runScript` with the custom config imported as well?

Reply:
👍 Can add that also I rewrote a version of runScript that allows you to run Scripts with a slightly different type signature. e.g. if you want your script to be able to return a Maybe, Either, List value etc. Essentially I wanted to get some Aeson/Lens stuff working (e.g. keep trying random filters until I had accessed the right part of a json data structure) on data I fetched from the db in a script: 

```
-- | Initializes IHP and then runs the script inside the framework context allows for Script to return arbitrary monad.
runCustomScript :: (Monad m) => ConfigBuilder -> ((?modelContext :: ModelContext, ?context :: FrameworkConfig) => IO (m a)) -> IO (m a)
runCustomScript configBuilder taskMain = withUtf8 do
    frameworkConfig@FrameworkConfig { environment, dbPoolIdleTime, dbPoolMaxConnections, databaseUrl, logger } <- buildFrameworkConfig configBuilder
    modelContext <- createModelContext dbPoolIdleTime dbPoolMaxConnections databaseUrl logger

    let ?modelContext = modelContext
    let ?context = frameworkConfig
    (taskMain `finally` cleanup logger)

runTestName :: (?modelContext :: ModelContext, ?context :: FrameworkConfig) => IO (Either String GoogleMaps.Result)
runTestName = do
    site <- fetchOne (query @Site |> filterWhere(#postcode, "XXXXXX"))
    let geoCodeResult = (eitherDecode $ cs (get #googleMapsJson site))
    pure geoCodeResult
    ``` 

This then lets you return data from the script.         
```    
  IHP> geoResult <- runCustomScript ihpDefaultConfig runTestName 
```    

There might already be a better way of getting Query results in ghci? But this is the most convenient way I could find rather than writing the full query + processing in the interpreter.

Reply:
This looks like something that rather should be a high level IHP function. Otherwise the `runCustomScript` relies on a lot of IHP internals and will break when we change those. Basically we need a `withIHP` function 👍

Reply:
we can have an example call to runScript with the custom config imported

Reply:
👍 looks like what I had in mind.

I'd actually make this function part of IHP to avoid B.C. issues in the future when IHP internals change.